src/utils/IO/branchdata_checks.jl

```diff
@@ -59,105 +59,114 @@ function sanitize_angle_limits!(line)
-    69.0 => (min = 12.0, max = 13.0),
-    138.0 => (min = 47.0, max = 52.0),
-    230.0 => (min = 134.0, max = 145.0),
-    345.0 => (min = 325.0, max = 425.0),
-    500.0 => (min = 850.0, max = 1075.0),
-    765.0 => (min = 2200.0, max = 2300.0),
-)
+"""
+Calculates the line rating based on the formula for the maximum transfer limit over an impedance
+"""
+function line_rating_calculation(l::Union{Line, MonitoredLine})
+    theta_max = max(abs(l.angle_limits.min), abs(l.angle_limits.max))
 
-# calculation from https://neos-guide.org/sites/default/files/line_flow_approximation.pdf
-function calculate_sil(line, basemva::Float64)
-    arc = get_arc(line)
-    #Assumess voltage at both ends of the arc is the same
-    vrated = get_base_voltage(get_to(arc))
-    zbase = vrated^2 / basemva
-    l = get_x(line) / (2 * pi * 60) * zbase
-    r = get_r(line) * zbase
-    c = sum(get_b(line)) / (2 * pi * 60 * zbase)
-    # A line with no C doesn't allow calculation of SIL
-    isapprox(c, 0.0) && return Inf
-    zc = sqrt((r + im * 2 * pi * 60 * l) / (im * 2 * pi * 60 * c))
-    sil = vrated^2 / abs(zc)
-    return sil
-end
+    g = l.r / (l.r^2 + l.x^2)
+    b = -l.x / (l.r^2 + l.x^2)
+    y_mag = sqrt(g^2 + b^2)
 
-function check_sil_values(line::Union{Line, MonitoredLine}, basemva::Float64)
-    arc = get_arc(line)
-    vrated = get_base_voltage(get_to(arc))
-    SIL_levels = collect(keys(SIL_STANDARDS))
-    rating = get_rating(line)
-    closestV = findmin(abs.(SIL_levels .- vrated))
-    closestSIL = SIL_STANDARDS[SIL_levels[closestV[2]]]
-    is_valid = true
+    from_voltage_limits = get_voltage_limits(get_arc(l).from)
+    to_voltage_limits = get_voltage_limits(get_arc(l).to)
 
-    # Assuming that the rate is in pu
-    if (rating >= closestSIL.max / basemva)
-        # rate outside of expected SIL range
-        sil = calculate_sil(line, basemva)
-        @warn "Rating $(round(rating*basemva; digits=2)) MW for $(line.name) is larger than the max expected in the range of $(closestSIL)." maxlog =
-            PS_MAX_LOG
-        is_valid = false
+    fr_vmin = isnothing(from_voltage_limits) ? 0.9 : from_voltage_limits.min
+    to_vmin = isnothing(to_voltage_limits) ? 0.9 : from_voltage_limits.min
+
+    c_max = sqrt(fr_vmin^2 + to_vmin^2 - 2 * fr_vmin * to_vmin * cos(theta_max))
+    new_rate = y_mag * max(fr_vmin, to_vmin) * c_max
+
+    return new_rate
+end
+
+function correct_rate_limits!(branch::Union{Line, MonitoredLine}, basemva::Float64)
+    theoretical_line_rate_pu = line_rating_calculation(branch)
+    for field in [:rating, :rating_b, :rating_c]
+        rating_value = getfield(branch, field)
+        if isnothing(rating_value)
+            @assert field ∈ [:rating_b, :rating_c]
+            continue
+        end
+        if rating_value < 0.0
+            @error "PowerSystems does not support negative line rates. $(field): $(rating)"
+            return false
```

Comment:
If we are not supporting this, shall we throw the error instead?

Reply:
if false is returned it fails after